packages/sync/src/connect-indexdb.js

```diff
@@ -20,12 +20,16 @@ import { IndexeddbPersistence } from 'y-indexeddb';
  * @return {Promise<() => void>} Promise that resolves when the connection is established.
  */
 export function connectIndexDb( objectId, objectType, doc ) {
-	const docName = `${ objectType }-${ objectId }`;
-	const provider = new IndexeddbPersistence( docName, doc );
+	if ( process.env.IS_GUTENBERG_PLUGIN ) {
```

Comment:
I don't think we should touch the sync package directly, the check should be on the import side instead

Reply:
Yes, the check could be added before `window.__experimentalEnableSync` checks in `@wordpress/core-data`. If everything works correctly, then in the https://github.com/WordPress/gutenberg/pull/54738#issuecomment-1731614338, you should see no size increase. At the moment, everything gets bundled:

<img width="531" alt="Screenshot 2023-09-22 at 20 38 16" src="https://github.com/WordPress/gutenberg/assets/699132/c37acf4a-e79a-4b42-9b59-4079218d9e05">

Reply:
In Gutenberg, the environment variable is enabled so I think you'll still see the bundle size increase but on WordPress Core no.

Reply:
@youknowriad, you are totally correct. It’s hard to navigate with all the env variables that depend on the context 🙈

Reply:
Updated!